awx/main/models/workflow.py

```diff
@@ -144,6 +146,18 @@ class WorkflowJobTemplateNode(WorkflowNodeBase):
         related_name='workflow_job_template_nodes',
         on_delete=models.CASCADE,
     )
+    identifier = models.CharField(
+        max_length=512,
+        default=uuid4,
+        blank=False,
+        help_text=_(
+            'An identifier for this node that is unique within its workflow. '
+            'It is copied to workflow job nodes corresponding to this node.'),
```

Comment:
Should you `db_index=True` this field?

Reply:
Also, do you want to put a unique constraint on this field so people don't provide duplicate values?

Reply:
Questions about db index...

 - Shouldn't it be indexed _together with_ the workflow (job or template)? It's not globally unique, and if you're doing a global filter, you're pretty much doing it wrong. Is indexing together more memory efficient than just making the field indexed?
 - index the WFJT node identifier, or the workflow job identifier, or both?

Also, is `name` not indexed?

https://github.com/ansible/awx/blob/d2998b42f169cec82e0d7a8d2890739237107ced/awx/main/models/base.py#L357-L360

that doesn't sound right

Reply:
> Also, is name not indexed?

I don't think we index names.

Reply:
> Shouldn't it be indexed together with the workflow (job or template)?

We could if we wanted to 🤷‍♂.  We just need _some_ index on this, given its intended use.

Reply:
yeah I don't think this should be a problem at all.

This will be adding an `index_together` in the exact same way as the `unique_together` for the WFJT nodes. for workflow job nodes, we would be adding `index_together`, where no unique constraint exists. There is _some_ concern about the time that will be incurred during the migration adding this. All pre-migration job nodes will have a blank identifier (timing impact unclear).

Also, both of these options I'm talking about are deprecated

https://docs.djangoproject.com/en/3.0/ref/models/options/

Given the small scope of the change here, it makes sense to take the time to move to the modern patterns, and I'll try to do that when I get back around to this branch.